src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/CosmosDBSourceConfig.java

```diff
@@ -0,0 +1,383 @@
+package com.microsoft.azure.cosmosdb.kafka.connect.source;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Width;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import com.microsoft.azure.cosmosdb.kafka.connect.TopicContainerMap;
+
+/**
+ * Contains settings for the Kafka ComsosDB Source Connector
+ */
+
+@SuppressWarnings ({"squid:S1854", "squid:S2160"})  // suppress unneeded int *groupOrder variables, equals method
+public class CosmosDBSourceConfig extends AbstractConfig {
+
+    static final String COSMOS_CONN_ENDPOINT_CONF = "connect.cosmosdb.connection.endpoint";
```

Comment:
properties like Endpoint, Key, Datbase, Container name are common across Sync & Source. and they're repeated now in ...SourceConfig.java and in ...SinkConfig.java. 

is it possible to have a CosmosDBConfig.java and have the Sink and Source config inherit from the common one? 

or is the thing that got us in to the mess last time?

Reply:
Not exactly, that can certainly be done.

Reply:
this would be nice to do. depending on effort involved.

Reply:
shouldn't be too bad, i can get it in this PR.